### PR TITLE
On click, a new layer connects to bottom of node if possible.

### PR DESCRIPTION
**Pull request by nu11us**
If you have concerns, it works on my local version.

### CODE REVIEW

File: ide/static/js/canvas.js [hunk 1]
```diff
@@ -70,6 +70,15 @@ class Canvas extends React.Component {
       this.props.changeNetStatus(false);
       // instance.repaintEverything();
     }
+    if(Object.keys(net).length>1){
+      const x1 = parseInt(net[`l${this.props.nextLayerId-2}`].state.top.split('px'));
+      const x2 = parseInt(net[`l${this.props.nextLayerId-1}`].state.top.split('px'));
+      if (x2-x1==100) {
```
Comment (utsavgarg) [inline ide/static/js/canvas.js]: change `100` to `80`

File: ide/static/js/content.js [hunk 2]
```diff
@@ -66,6 +67,8 @@ class Content extends React.Component {
     this.zooModal = this.zooModal.bind(this);
     this.modalContent = null;
     this.modalHeader = null;
+    this.clicker = false;
```
Comment (utsavgarg) [inline ide/static/js/content.js]: Where is this being used?

File: ide/static/js/content.js [hunk 3]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
     this.modalContent = <ModelZoo importNet={this.importNet}/>;
     this.openModal();
   }
+  
+  handleClick(event) {
+    this.placeholder = false;
```
Comment (utsavgarg) [inline ide/static/js/content.js]: Remove this

File: ide/static/js/content.js [hunk 4]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
...
+    event.preventDefault();
+
+    const net = this.state.net;
+    const id = event.target.id;
+    const prev = net[`l${this.state.nextLayerId-1}`];
+    const next = data[id];
+    const zoom = instance.getZoom();    
+    const layer = {};
+    let phase = this.state.selectedPhase;
+
```
Comment (utsavgarg) [inline ide/static/js/content.js]: Need to add a case to handle situation when there is no layer, can do something like this
```
// To handle case when no layer has been added
    if (Object.keys(net).length==0){
      layer.connection = { input: [], output: [] };
      layer.info = {
            type: id.toString(),
            phase,
            class: ''
          }
      layer.params = {'endPoint': [next['endpoint'], false]};        
      Object.keys(next.params).forEach(j => {
        layer.params[j] = [next.params[j].value, false];
      });    
      layer.props = JSON.parse(JSON.stringify(next.props))
      const height = Math.round(0.05*window.innerHeight, 0);
      const width = Math.round(0.35*window.innerWidth, 0);
      var top = height + Math.ceil(41-height);
      var left = width;
      layer.state = {
            top: `${top}px`,
            left: `${left}px`,
            class: '' 
          }
      layer.props.name = `${next.name}${this.state.nextLayerId}`;          
      this.addNewLayer(layer);
      this.clicker = true;
    }
```

File: ide/static/js/content.js [hunk 5]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
...
+    const net = this.state.net;
+    const id = event.target.id;
+    const prev = net[`l${this.state.nextLayerId-1}`];
+    const next = data[id];
+    const zoom = instance.getZoom();    
+    const layer = {};
+    let phase = this.state.selectedPhase;
+    
+    if (this.state.nextLayerId>0) {
```
Comment (utsavgarg) [inline ide/static/js/content.js]: need to make this `else if`

File: ide/static/js/content.js [hunk 6]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
...
+      if (prev.params.endPoint[0].src == "Bottom"){
```
Comment (utsavgarg) [inline ide/static/js/content.js]: change to `if (data[prev.info.type].endpoint.src == "Bottom"){`

File: ide/static/js/content.js [hunk 7]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
...
+        if (next.endpoint.trg == "Top") {
+
+          layer.connection = { input: [], output: [] };
+          layer.info = {
+            type: id.toString(),
+            phase,
+            class: ''
+          }
+          layer.params = {}
+          layer.params['endPoint'] = [next['endpoint'], false];
```
Comment (utsavgarg) [inline ide/static/js/content.js]: remove this

File: ide/static/js/content.js [hunk 8]
```diff
@@ -680,6 +683,47 @@ class Content extends React.Component {
...
+            class: ''
+          }
+          layer.params = {}
+          layer.params['endPoint'] = [next['endpoint'], false];          
+          Object.keys(next.params).forEach(j => {
+            layer.params[j] = [next.params[j].value, false];
+          });    
+          layer.props = JSON.parse(JSON.stringify(next.props))
+          layer.state = {
+            top: `${(parseInt(prev.state.top.split('px')[0])/zoom + 100)}px`,
```
Comment (utsavgarg) [inline ide/static/js/content.js]: change `100` to `80`

File: ide/static/js/content.js [hunk 9]
```diff
@@ -728,6 +774,7 @@ class Content extends React.Component {
             error={this.state.error}
             dismissError={this.dismissError}
             addError={this.addError}
+            clicker={this.clicker}
```
Comment (utsavgarg) [inline ide/static/js/content.js]: I see this being passed to `canvas` but don't see it being used there.

File: ide/static/js/content.js (hunk 2 above)
Comment (nu11us) [reply ide/static/js/content.js]: Leftover that I forgot to delete.

Comment (utsavgarg) [reply ide/static/js/content.js]: Still haven't deleted this?

**Pull request merged**